extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/health/KafkaStreamsStateHealthCheck.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.kafka.streams.runtime.health;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.Liveness;
+
+import io.quarkus.kafka.streams.runtime.KafkaStreamsTopologyManager;
+
+@Liveness
+@ApplicationScoped
+public class KafkaStreamsStateHealthCheck implements HealthCheck {
+
+    @Inject
+    private KafkaStreamsTopologyManager manager;
+
+    @Override
+    public HealthCheckResponse call() {
+        return HealthCheckResponse.named("Kafka Streams state health check")
+                .state(manager.getStreams().state().isRunning())
+                .withData("state", manager.getStreams().state().name())
+                .build();
+    }
```

Comment:
You should add this to a try/catch block and generates a DOWN liveness in case of any exception